chat.js

```diff
@@ -350,6 +353,38 @@ class CommandContext {
 		}
 		return true;
 	}
+	checkR9k(user, room, message) {
+		let checksum = Chat.md5(toId(message)).substr(0, 16);
+
+		if (!room.r9kTempCache) room.r9kTempCache = new Set();
+
+		if (room.r9kTempCache.has(checksum) || (room.r9kPermanentCache && room.r9kPermanentCache.has(checksum))) {
+			// If the phrase is in the temporary cache, add it to the permanent cache
+			if (!room.r9kPermanentCache) room.r9kPermanentCache = new Set();
+			if (!room.r9kPermanentCache.has(checksum)) {
+				Chat.r9kAppendPermanent(room, checksum);
```

Comment:
So messages will have to be said at least twice without a restart in between to be properly added? Why?